src/Components/Locations.js

```diff
@@ -0,0 +1,61 @@
+// "https://ghibliapi.herokuapp.com/locations/"
+import axios from "axios";
+import { Component } from "react";
+
+class Locations extends Component {
+  state = { places: [], showLoca: false };
+
+  fetchLocations = async () => {
+    try {
+      const res = await axios.get("https://ghibliapi.herokuapp.com/locations/");
+      this.setState({ places: res.data.map((location) => {
+          location = {name: location.name, climate: location.climate, terrain: location.terrain}
+      }) });
+        // debugger
+    } catch (error) {
+      this.setState({ places: [] });
+    }
+  };
+
+  componentDidMount() {
+    this.fetchLocations();
+  }
+ 
+
+  handleClick = (e) => {
+    // this.setState((prevState) => {
+    //   return { showLoca: !prevState.showLoca };
+    // });
```

Comment:
I prefer the commented out code here. 

Because you're using the current state to derive the next state we should use the callback.